File: src/test/java/pl/mjedynak/idea/plugins/builder/factory/PackageChooserDialogFactoryTest.java
package pl.mjedynak.idea.plugins.builder.factory;

import static org.assertj.core.api.Assertions.assertThat;
import static org.mockito.BDDMockito.given;

import com.intellij.ide.util.PackageChooserDialog;
import com.intellij.openapi.project.Project;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;
import org.mockito.Mock;
import org.mockito.MockedStatic;
import org.mockito.junit.jupiter.MockitoExtension;

@ExtendWith(MockitoExtension.class)
public class PackageChooserDialogFactoryTest {

    @Mock
    private MockedStatic<PackageChooserDialogFactory> packageChooserDialogFactory;

    @Mock
    private Project project;

    @Mock
    private PackageChooserDialog packageChooserDialog;

    @Test
    void shouldCreatePackageChooserDialogWithPassedTitle() {
        // given
        String title = "title";
        given(packageChooserDialog.getTitle()).willReturn(title);
        packageChooserDialogFactory
                .when(() -> PackageChooserDialogFactory.getPackageChooserDialog(title, project))
                .thenReturn(packageChooserDialog);

        // when
        PackageChooserDialog result = PackageChooserDialogFactory.getPackageChooserDialog(title, project);

        // then
        assertThat(result).isEqualTo(packageChooserDialog);
        assertThat(result.getTitle()).isEqualTo(title);
    }
}

File: src/test/kotlin/io/github/kingg22/kobra/builder/factory/PackageChooserDialogFactoryTest.kt
package io.github.kingg22.kobra.builder.factory

import com.intellij.ide.util.PackageChooserDialog
import com.intellij.openapi.project.Project
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.extension.ExtendWith
import org.mockito.BDDMockito.given
import org.mockito.Mock
import org.mockito.MockedStatic
import org.mockito.junit.jupiter.MockitoExtension

@ExtendWith(MockitoExtension::class)
class PackageChooserDialogFactoryTest {
    @Mock
    private lateinit var packageChooserDialogFactory: MockedStatic<PackageChooserDialogFactory>

    @Mock
    private lateinit var project: Project

    @Mock
    private lateinit var packageChooserDialog: PackageChooserDialog

    @Test
    fun shouldCreatePackageChooserDialogWithPassedTitle() {
        // given
        val title = "title"
        given(packageChooserDialog.title).willReturn(title)
        packageChooserDialogFactory.`when`<PackageChooserDialog> {
            PackageChooserDialogFactory.getPackageChooserDialog(title, project)
        }.thenReturn(packageChooserDialog)

        // when
        val result = PackageChooserDialogFactory.getPackageChooserDialog(title, project)

        assertThat(result).isEqualTo(packageChooserDialog)
        assertThat(result.title).isEqualTo(title)
    }
}
